src/decoder/plugins/SidplayDecoderPlugin.cxx

```diff
@@ -437,51 +439,83 @@ sidplay_file_decode(DecoderClient &client, Path path_fs)
 	} while (cmd != DecoderCommand::STOP);
 }
 
+static AllocatedString<char>
+Windows1252ToUTF8(const char *s)
+{
+#ifdef HAVE_ICU_CONVERTER
+	try {
+		std::unique_ptr<IcuConverter>
+			converter(IcuConverter::Create("windows-1252"));
+
+		return converter->ToUTF8(s);
+	} catch (...) { }
+#endif
+
+	/*
+	 * Fallback to not transcoding windows-1252 to utf-8, that may result
+	 * in invalid utf-8 unless nonprintable characters are replaced.
+	 */
+	AllocatedString<char> t = AllocatedString<char>::Duplicate(s);
+
+	for (size_t i = 0; t[i] != AllocatedString<char>::SENTINEL; i++)
+		if (!std::isprint(t[i]))
```

Comment:
`isprint()` is a bad choice, because it is locale-dependent and may return true on non-ASCII characters. It does many things, but prevent invalid UTF-8 sequences is not what it does.
What you could use is `IsPrintableASCII()` from `util/CharUtil.hxx` which guarantees that it's printable *and* valid UTF-8 (because 7-bit ASCII is a subset of UTF-8).

Reply:
Hmm... I forgot about the locale stuff. `IsPrintableASCII` looks no good since `0x7f` is DEL, which isn't printable, so I think that must be corrected first:

```
constexpr bool
IsPrintableASCII(char ch) noexcept
{
	return (signed char)ch >= 0x20;
}
```

Reply:
You're right, this function should probably be corrected or renamed. Let's not care for that right now.